### ISSUE TITLE
Thunderbird 128

### ISSUE THREAD

**Issue by Janovlk** [jobisoft/EAS-4-TbSync#267]
Hello,

do you intend to make the add-on compatible with Thunderbird v. 128?

Thanks for your work,

PP

**Comment by spiregarden** [comment]
I'd love to find out too. TB 128 was pushed by my distro, so if TbSync is no longer maintained, I see only two options:
1) Downgrade TB to 115 (remember to make backups first)
2) Beta / Daily version of TB 128.1. According to [this](https://blog.thunderbird.net/2024/07/welcome-to-thunderbird-128-nebula/) post it may have experimental Exchange integration:
_"For advanced users who want to help test our initial implementation of Exchange (currently limited to Mail), it is now available in our Daily and Beta builds. This Wiki page has more information as well as instructions for enabling it. While we definitely welcome your testing and feedback, please keep in mind this feature is currently experimental, and you may run into unexpected behavior or errors."_

**Comment by jobisoft** [comment]
The update is being worked on. May I ask which distribution pushed you onto TB128 at this early stage? The Thunderbird team does not push the update for at least a another month.

In the long run, the built-in Exchange support (EWS) will also do calendars and contacts, but the current experimental code only does email. I *think* they will get there in 2025.

TbSync will still be needed for users who must use EAS.

**Comment by Janovlk** [comment]
I deliberately upgraded to v. 128. The current version(s) has a bug in displaying the list of events. But I successfully downgraded back to v. 115 afterwards.

**Comment by jobisoft** [comment]
I try to release the update this week.

**Comment by Nils-TUD** [comment]
> The update is being worked on. May I ask which distribution pushed you onto TB128 at this early stage? The Thunderbird team does not push the update for at least a another month.

Flathub already updated to TB128.

**Comment by jobisoft** [comment]
> > The update is being worked on. May I ask which distribution pushed you onto TB128 at this early stage? The Thunderbird team does not push the update for at least a another month.
> 
> Flathub already updated to TB128.

Thanks for the feedback, I did not know that.

**Comment by brtptrs** [comment]
Ubuntu 24.04 also pushed TB128

Thanks for the great work

**Comment by spiregarden** [comment]
Thank you @jobisoft for the great news and kudos for those who's sacrificing your time to make the update. Much appreciated!

In my case, Ubuntu 22.04 pushed to TB128 (snap).

Is there any thread about the progress of the update? (Just to assess whether I should downgrade meanwhile)

**Comment by jobisoft** [comment]
There are no major issues with esr 128, so it is ok to be used. They just do not want to push it to all users at once but use the next 2 month for additional testing. Addon developers also use these two month for updating their extensions.

**Comment by IpeTek** [comment]
Thanks for your consistent work along all these years! It's a perfect time for a donation, once again.

**Comment by axelrathey** [comment]
My TB115 was today automatically updated to TB128. Now, EAS-4-TbSync is deactivated. There is no version of this add-on for TB128 yet. Please update it now for TB128.

**Comment by bikerflyer** [comment]
TB128 also default on Mint 22 (Ubuntu 24.04)

**Comment by scylla999** [comment]
I'm willing to try Beta version of this extension for  TB128 support when it becomes available.   I've installed Beta version of TBSync.

**Comment by soumitrachatterjee** [comment]
> I'm willing to try Beta version of this extension for TB128 support when it becomes available. I've installed Beta version of TBSync.

+1

**Comment by soumitrachatterjee** [comment]
> My TB115 under Windows 11 was today automatically updated to TB128. Now, EAS-4-TbSync is deactivated. There is no version of this add-on for TB128 yet. Please update it now for TB128.

+1

**Comment by jdfoote** [comment]
@jobisoft Sorry to pester but is there an ETA for this? I'm trying to decide whether to revert back to an earlier Thunderbird or to wait.

Also, if there are things that a novice JS programmer could do to help, I'd be happy to do so. Thanks!!

**Comment by dr-hildi** [comment]
I got an auto-update on macOS this week to TB 128.0.1esr and 128.1.0esr today.

The 128 update disabled the plugin.

**Comment by soumitrachatterjee** [comment]
> I'm willing to try Beta version of this extension for TB128 support when it becomes available. I've installed Beta version of TBSync.

+1

**Comment by DanWiseProgramming** [comment]
Quick tip for those in a pinch:
I manually patched the version constraints for both the TBSync and the EAS 4 TBSync provider add-ons so TB 128 does not reject them, installed them from file and they appear to work correctly. E-mail send/receive and calendar sync all appear to work well.

What does not work correctly is the registration of the provider. This means that you cannot see and change your Exchange account information inside the add-on. Since I only upgraded and don't need to change any settings, I've left it as is until we get the next release of these add-ons or the official Exchange support from Thunderbird.

**Comment by Giermann** [comment]
@DanWiseProgramming what version did you patch to be compatible?
I just downloaded v4.8 of TbSync and EAS 4 TbSync (along the Google-4-TbSync 0.6.0), patched them to be compatible and installed them.
No error, no warning, the TbSync icon re-appeared but when I try to open TbSync dialog, nothing happens but a short flash of the main window. Also calendar sync does not seem to work, as recently added events still don't show up...

**Comment by DanWiseProgramming** [comment]
@Giermann I'll check tomorrow and let you know.

**Comment by jobisoft** [comment]
Beta versions for 128. Please backup your data before testing.

- [TbSync-beta-4.10.zip](https://github.com/user-attachments/files/16653921/TbSync-beta-4.10.zip)
- [EAS-4-TbSync-beta.4.9.zip](https://github.com/user-attachments/files/16653926/EAS-4-TbSync-beta.4.9.zip)

**Comment by ravensorrow** [comment]
> Beta versions for Thunderbird 128. Please backup your data before testing.
> 
>     * [TbSync-beta-4.10.zip](https://github.com/user-attachments/files/16653921/TbSync-beta-4.10.zip)
> 
>     * [EAS-4-TbSync-beta.4.9.zip](https://github.com/user-attachments/files/16653926/EAS-4-TbSync-beta.4.9.zip)

[Google-4-TbSync](https://addons.thunderbird.net/en-CA/thunderbird/addon/google-4-tbsync/) is a required dep that is not currently compatible w/ TB128.

![Google-4-TbSync not compatible (1)](https://github.com/user-attachments/assets/6746503c-3ab0-4ff7-bf23-45a6851b6c71)
![Google-4-TbSync not compatible](https://github.com/user-attachments/assets/3ac98ffc-ad8d-4b00-b23f-c3f36052dd79)

**Comment by Giermann** [comment]
> Beta versions for Thunderbird 128. Please backup your data before testing.
> 
>     * [TbSync-beta-4.10.zip](https://github.com/user-attachments/files/16653921/TbSync-beta-4.10.zip)
> 
>     * [EAS-4-TbSync-beta.4.9.zip](https://github.com/user-attachments/files/16653926/EAS-4-TbSync-beta.4.9.zip)

Working for me!
Thanks for your work!!

Of course it is not (yet) syncing my Google account because of the missing Google-4-TbSync, but that's out of scope here.

**Comment by Nils-TUD** [comment]
Works for me as well. Thanks a lot!

**Comment by jobisoft** [comment]
Note: Do not use the beta versions in TB 115. I have to mark them incompatible with TB115.

**Comment by level420** [comment]
Works for me. Only tested already configured accounts. Thanks allot!

**Comment by jim-davenport** [comment]
It's working for me. Thank you!

**Comment by myssv** [comment]
Thanks a lot for your work !!!

**Comment by smmehta1822** [comment]
Thanks a lot and it works.

**Comment by soumitrachatterjee** [comment]
> Beta versions for Thunderbird 128. Please backup your data before testing.
> 
>     * [TbSync-beta-4.10.zip](https://github.com/user-attachments/files/16653921/TbSync-beta-4.10.zip)
> 
>     * [EAS-4-TbSync-beta.4.9.zip](https://github.com/user-attachments/files/16653926/EAS-4-TbSync-beta.4.9.zip)

Works flawlessly! Thanks!

**Comment by Giermann** [comment]
There's a little issue with the provided beta versions: on every sync interval, a window "TbSync" opens and automatically closes immediately. This makes my screen flicker from time to time - is this intended or not known yet?

**Comment by jobisoft** [comment]
The TbSync add-on itself should now auto-update on Thunderbird 128 to the released version 4.11:
https://addons.thunderbird.net/addon/tbsync/versions/

The Google-4-TbSync provider for Thunderbird 128 using Google's People API is listed as well:
https://addons.thunderbird.net/addon/google-4-tbsync/versions/

My EAS provider still needs some time, new betas are published here.

**Comment by brtptrs** [comment]
I'm seeing the same as @Giermann in comment https://github.com/jobisoft/EAS-4-TbSync/issues/267#issuecomment-2296734198
It seems to be the M365 authentication screen that pops up. 
Three times today, TbSync seems to have lost the MS credentials, and I had to repeat the authentication. I'm not quite sure, but I think it happened after restarting Thunderbird.
TB is running as a snap on Ubuntu 2024

**Comment by jobisoft** [comment]
Beta 4.11 of the EAS provider has the following changes:

- sandbox http requests (isolate cookies per TbSync account), which could help with lost credentials if multiple O365 accounts are synced, it definitely allows me to sync an outlook.com account and a hotmail account in parallel
- setup wizard adjusts size if error message needs more space
- sets min requirement back to 128

[EAS-4-TbSync-beta-4.11.zip](https://github.com/user-attachments/files/16663990/EAS-4-TbSync-beta-4.11.zip)

@brtptrs and @Giermann Does this fix your issues? It could still happen once.

**Comment by brtptrs** [comment]
Thank you, I've installed 4.11 and will keep an eye on it.
I have not yet seen the pop-up since upgrading, but it's only been running for a few minutes.

**Comment by achmafooma** [comment]
Thank you so much for the update! Running the beta since this morning. Was having the occasional re-auth issue but hasn't happened since the latest update. Testing on Windows & Mac, both working fine so far.

**Comment by pzia** [comment]
Testing now EAS on Linux (Ubuntu 24.4) with the new autoupdated TbSync, working fine so far.

**Comment by ghost-from-the-past** [comment]
Tested the beta 4.11 - TB snap on Ubuntu 24.4, works fine (outlook.office360.com).
I needed to re-auth, for this I have 2FA with a yubikey which didn't work (may be a problem of the TB snap closing the access) then I used TOTP without problem.

**Comment by ghost-from-the-past** [comment]
I tested also in my MacOS.
The YubiKey worked in this case, so the previous should be a snap access problem.
Here I did the migration from 115 to 128 and then added the beta extension.
At the synchronization I got a 16 synchro error (if helps I only synchro Calendar and Birthdays).
I just disable and re-enable the account at the "TbSync account manager" and at the resync no error this time.

**Comment by jobisoft** [comment]
@ghost-from-the-past : Thanks for your tests and feedback.

**Comment by Giermann** [comment]
> @brtptrs and @Giermann Does this fix your issues? It could still happen once.

No, unfortunately not - I still see the auth window on almost every sync for about half a second, then closing itself.
Same issue like in beta 4.9

**Comment by Cyanic** [comment]
I'm using Fedore KDE 40 happy to test this out. I'm just not sure how to test it? Is it just a matter of unzipping the files and moving them somewhere? Or what's the process of testing the beta?

**Comment by jobisoft** [comment]
> I'm using Fedore KDE 40 happy to test this out. I'm just not sure how to test it? Is it just a matter of unzipping the files and moving them somewhere? Or what's the process of testing the beta?

1. Download the zip file (do not unpack it)
2. Open Thunderbird and open the add-on manager
3. Click on the gear icon and select "install add-on from file" and select the zip file

**Comment by smmehta1822** [comment]
> > I'm using Fedore KDE 40 happy to test this out. I'm just not sure how to test it? Is it just a matter of unzipping the files and moving them somewhere? Or what's the process of testing the beta?
> 
>     1. Download the zip file (do not unpack it)
> 
>     2. Open Thunderbird and open the add-on manager
> 
>     3. Click on the gear icon and select "install add-on from file" and select the zip file

I followed the same steps and still see a Auth pop up. I am using TB 128 on a MacOS 14.6.1
Using TBSync 4.11 and 
EAS-4-TBSync - 4.11

**Comment by jobisoft** [comment]
> > > I'm using Fedore KDE 40 happy to test this out. I'm just not sure how to test it? Is it just a matter of unzipping the files and moving them somewhere? Or what's the process of testing the beta?
> > 
> > 
> > ```
> > 1. Download the zip file (do not unpack it)
> > 
> > 2. Open Thunderbird and open the add-on manager
> > 
> > 3. Click on the gear icon and select "install add-on from file" and select the zip file
> > ```
> 
> I followed the same steps and still see a Auth pop up. I am using TB 128 on a MacOS 14.6.1 Using TBSync 4.11 and EAS-4-TBSync - 4.11

What type of account? Can you try to setup up the account again in TbSync?

**Comment by 9p4** [comment]
I have noticed that the Microsoft login page pops up very briefly (half a second or less), and then disappears as it logs itself in.

**Comment by jobisoft** [comment]
> I have noticed that the Microsoft login page pops up very briefly (half a second or less), and then disappears as it logs itself in.

That should happen only once or twice, but not on every sync. If it does happen on every sync, I would kindly ask to setup TbSync on a fresh profile, to rule out any profile-dependent quirks (we can try to resolve these in a second step).

**Comment by 9p4** [comment]
It has only happened once so far, not on every sync.

**Comment by jobisoft** [comment]
> It has only happened once so far, not on every sync.

Once or twice is expected, I moved the connection into an isolation wrapper, which means it does not have access to the old auth cookies, but still knows the auth details. So it has to re-auth once.

**Comment by jobisoft** [comment]
If there are no major complaints, I am going to release ESR to the public today.

**Comment by Giermann** [comment]
I agree to release to public. I did not manage to try with a clean profile yet, but have to revert my observation:
It does not seem to happen on every sync, but also after a restart of TB it happened 2 or 3 times today.

I'll keep an eye on it and we should continue this on a different issue.

**Comment by soumitrachatterjee** [comment]
It does happen repeatedly, though I haven't had a chance to try with a clean profile yet.

**Comment by jobisoft** [comment]
Released!

**Comment by myssv** [comment]
When I come back from power save mode, I often got this message:

![grafik](https://github.com/user-attachments/assets/834d79f0-871d-404a-abeb-21f4f07ec071)

a retry and all is good

**Comment by achmafooma** [comment]
> Released!

Thank you for all your hard work getting this done, John. Great job!

**Comment by henrygouk** [comment]
With the latest beta (and also release), I still get periodic reauth requests. It doesn't ask for the username/password; it only asks for the multi-factor authentication code.

**Comment by jobisoft** [comment]
> With the latest beta (and also release), I still get periodic reauth requests. It doesn't ask for the username/password; it only asks for the multi-factor authentication code.

Also with fresh TbSync account or fresh Thunderbird profile?

**Comment by henrygouk** [comment]
Just tried with a fresh Thunderbird profile and TbSync account. It works fine for about an hour, then I start getting the 2FA popups again.

**Comment by jobisoft** [comment]
Can you screenshot the popup (and remove private info if needed)?

**Comment by henrygouk** [comment]
![image](https://github.com/user-attachments/assets/7196b78d-a476-4547-b6fe-58a93746df75)

**Comment by jobisoft** [comment]
Can you  create a new issue for this? I will try to reproduce this.
